fish-rust/src/fd_monitor.rs

```diff
@@ -0,0 +1,573 @@
+use std::os::fd::{AsRawFd, RawFd};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+
+use self::fd_monitor::{c_void, new_fd_event_signaller, FdEventSignaller, ItemWakeReason};
+use crate::fd_readable_set::FdReadableSet;
+use crate::fds::AutoCloseFd2;
+use crate::ffi::void_ptr;
+use crate::flog::FLOG;
+use crate::wutil::perror;
+use cxx::UniquePtr;
+
+#[cxx::bridge]
+mod fd_monitor {
+    /// Reason for waking an item
+    #[repr(u8)]
+    #[cxx_name = "item_wake_reason_t"]
+    enum ItemWakeReason {
+        /// The fd became readable (or was HUP'd)
+        Readable,
```

Comment:
I've been keeping enum variant names but I guess we can change them too

Reply:
I was going to typedef/using/define it but it was just a single call site so it was just easier to change it cpp side.